Description: Update CMake config for debian package
 A few updates to the CMakeLists.txt file to be more relevant for the
 Debian package.
Author: Matthew Newton <matthew-git@newtoncomputing.co.uk>
Last-Update: 2019-03-04

Index: libkqueue-2.3.1/CMakeLists.txt
===================================================================
--- libkqueue-2.3.1.orig/CMakeLists.txt
+++ libkqueue-2.3.1/CMakeLists.txt
@@ -15,7 +15,7 @@
 #
 
 # Note: In order for RPM packaging to work correctly version >= 3.8 is required
-cmake_minimum_required(VERSION 3.8.0)
+cmake_minimum_required(VERSION 3.7.2)
 cmake_policy(SET CMP0063 OLD)
 
 project(libkqueue VERSION 2.3.1 LANGUAGES C)
@@ -139,20 +139,26 @@
 set_target_properties(kqueue PROPERTIES DEBUG_POSTFIX "d")
 set_target_properties(kqueue PROPERTIES SOVERSION 0)
 
+add_library(kqueue_static STATIC ${LIBKQUEUE_SOURCES} ${LIBKQUEUE_HEADERS})
+set_target_properties(kqueue_static PROPERTIES DEBUG_POSTFIX "d" OUTPUT_NAME kqueue)
+
 if(WIN32)
   target_compile_definitions(kqueue PRIVATE _USRDLL;_WINDLL)
   target_compile_definitions(kqueue PRIVATE _CRT_SECURE_NO_WARNINGS)
 else()
   target_compile_definitions(kqueue PRIVATE _XOPEN_SOURCE=600)
+  target_compile_definitions(kqueue_static PRIVATE _XOPEN_SOURCE=600)
 endif()
 
 target_include_directories(kqueue PRIVATE include)
 if(NOT WIN32)
   target_include_directories(kqueue PRIVATE src/common)
+  target_include_directories(kqueue_static PRIVATE src/common include)
 endif()
 
 if(CMAKE_C_COMPILER_ID MATCHES GNU)
   target_compile_options(kqueue PRIVATE -Wall -Werror)
+  target_compile_options(kqueue_static PRIVATE -Wall -Werror)
 endif()
 if(MINGW AND CMAKE_C_COMPILER_ID MATCHES GNU)
   target_compile_options(kqueue PRIVATE -march=i486)
@@ -162,6 +168,7 @@
   target_link_libraries(kqueue PRIVATE Ws2_32)
 endif()
 target_link_libraries(kqueue PRIVATE Threads::Threads)
+target_link_libraries(kqueue_static PRIVATE Threads::Threads)
 
 if(ENABLE_TESTING)
   add_subdirectory(test)
@@ -182,10 +189,10 @@
 install(FILES
           "include/sys/event.h"
         DESTINATION
-          "${CMAKE_INSTALL_FULL_INCLUDEDIR}/sys"
+          "${CMAKE_INSTALL_FULL_INCLUDEDIR}/kqueue/sys"
         COMPONENT headers)
 install(TARGETS
-          kqueue
+          kqueue kqueue_static
         DESTINATION
           "${CMAKE_INSTALL_FULL_LIBDIR}${LIB_SUFFIX}"
         COMPONENT libraries)
